src/server/scripts/Outland/BlackTemple/boss_illidan.cpp

```diff
@@ -979,6 +989,7 @@ class boss_illidan_stormrage : public CreatureScript
                         break;
                     case EVENT_DEFEATED_TEXT:
                         Talk(SAY_ILLIDAN_DEFEATED);
+                        me->GetMap()->SetZoneMusic(BLACK_TEMPLE_ZONE_ID, EVENT_BT_ARRIVAL_WALK_HERO_1_SOUND_ID);
```

Comment:
SAY_ILLIDAN_DEFEATED already has a sound (check https://github.com/TrinityCore/TrinityCore/pull/18963/files#diff-d0664a82257023734a5dc1395ed99979R182 )
This case is zone music change xd